### ISSUE TITLE
Table - implicit_returning not compatible with extend_existing

### ISSUE THREAD

**Issue by jvanasco** [sqlalchemy/sqlalchemy#7295]
### Describe the bug

While working on some test cases to deprecate `implicit_returning` on create_engine, I tried to re-define a table.

### To Reproduce

```python
from sqlalchemy import Column
from sqlalchemy import MetaData
from sqlalchemy import Table
from sqlalchemy import String
from sqlalchemy.exc import InvalidRequestError

metadata = MetaData()

Table(
    "a",
    metadata,
    Column("foo", String, primary_key=True),
    implicit_returning=False,
)

try:
    Table(
        "a",
        metadata,
        Column("foo", String, primary_key=True),
        implicit_returning=True,
    )
    raise ValueError("Unexpected Fail")
except InvalidRequestError as exc:
    pass

Table(
    "a",
    metadata,
    Column("foo", String, primary_key=True),
    implicit_returning=True,
    extend_existing=True,
)
```

### Error

/Users/jvanasco/Desktop/implicit_extend.py:28: SAWarning: Can't validate argument 'implicit_returning'; can't locate any SQLAlchemy dialect named 'implicit'

### Versions

- OS:
- Python:
- SQLAlchemy:
- Database:
- DBAPI (eg: psycopg, cx_oracle, mysqlclient):

**Comment by zzzeek** [comment]
here is the fix.  this needs a test in test/sql/test_metadata.py if you want to provide a gerrit.  i have it here taking the new value of implicit returning and using it directly, whether it's True or False, overwriting the previous value. the test should exercise that too.   look in test_metadata.py for other tests that are exercising the implicit_returning flag, there are a bunch.

```diff
diff --git a/lib/sqlalchemy/sql/schema.py b/lib/sqlalchemy/sql/schema.py
index e45e22564d..14346cf609 100644
--- a/lib/sqlalchemy/sql/schema.py
+++ b/lib/sqlalchemy/sql/schema.py
@@ -756,6 +756,10 @@ class Table(DialectKWArgs, SchemaItem, TableClause):
         kwargs.pop("extend_existing", False)
         kwargs.pop("keep_existing", False)
 
+        self.implicit_returning = kwargs.pop(
+            "implicit_returning", self.implicit_returning
+        )
+
         if schema and schema != self.schema:
             raise exc.ArgumentError(
                 "Can't change schema of existing table from '%s' to '%s'",
```

**Comment by jvanasco** [comment]
Will do!

**Comment by sqla-tester** [comment]
**jonathan vanasco** has proposed a fix for this issue in the **main** branch:

**Fixes: #7295** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3287

**Comment by sqla-tester** [comment]
**jonathan vanasco** has proposed a fix for this issue in the **rel_1_4** branch:

**Fixes: #7295** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3291
